### ISSUE TITLE
Mapbox Android Demo app StyleLineIdentityPropertyActivity cause incorrect map rendering

### ISSUE THREAD

**Issue by Guardiola31337** [mapbox/mapbox-gl-native#11686]
While testing latest version of [Mapbox Android Demo app](https://github.com/mapbox/mapbox-android-demo) found 👇

**Platform:** Android 8.1.0 Pixel XL
**Mapbox SDK version:** `6.0.0-beta.7`

### Steps to trigger behavior

 1. Mapbox Android Demo app
 2. Data-driven styling
 3. Launch [`Style lines using an identity property function` activity](https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/StyleLineIdentityPropertyActivity.java)
 4. Observe unexpected rendering issues when zooming on the map 👀

![rendering_issues_boba_beta_7](https://user-images.githubusercontent.com/1668582/38802925-f88828e0-416d-11e8-9de8-abd3afe6f3a6.gif)

**Comment by langsmith** [comment]
To be clear, it's that green flashing line/box. I didn't initially see it when @Guardiola31337 showed me

**Comment by Guardiola31337** [comment]
> To be clear, it's that green flashing line/box.

Yeah those are the rendering artifacts seen above (sorry if I didn't make myself clear).

**Comment by cvance** [comment]
I see similar issues using data-driven styling on mapbox 6.1.1 in our production app. I'll investigate further tomorrow but I'm seeing lots of artifacts.
